docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,285 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.resourcedetection/
+labels:
+  stage: beta
+title: otelcol.processor.resourcedetection
+description: Learn about otelcol.processor.resourcedetection
+---
+
+# otelcol.processor.resourcedetection
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" version="<AGENT VERSION>" >}}
+
+`otelcol.processor.resourcedetection` detects resource information and adds resource attributes to logs, 
+metrics, and traces telemetry data.
+
+{{% admonition type="note" %}}
+`otelcol.processor.resourcedetection` is a wrapper over the upstream
+OpenTelemetry Collector Contrib `resourcedetection` processor. If necessary, 
+bug reports or feature requests will be redirected to the upstream repository.
+{{% /admonition %}}
+
+You can specify multiple `otelcol.processor.resourcedetection` components by giving them
+different labels.
+
+## Usage
+
+```river
+otelcol.processor.resourcedetection "LABEL" {
+  output {
+    logs    = [...]
+    traces  = [...]
+  }
+}
+```
+
+## Arguments
+
+`otelcol.processor.resourcedetection` supports the following arguments:
+
+Name        | Type           | Description                                                                            | Default | Required
+----------- | -------------- | -------------------------------------------------------------------------------------- |-------- | --------
+`detectors` | `list(string)` | An ordered list of named detectors which should be ran to detect resource information. |         | yes
+`override`  | `bool`         | Configures whether existing resource attributes should be overriden or preserved.      | `true`  | no
+`timeout`   | `duration`     | Timeout by which all specified detectors must complete.                                | `"5s"`  | no
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`otelcol.processor.resourcedetection`:
+
+Hierarchy                              | Block                                                       | Description                                       | Required
+-------------------------------------- | ----------------------------------------------------------- | ------------------------------------------------- | --------
```

Comment:
I'd like to rename this section to "Configuration" and remove the table with blocks. In its place I could put a table with "root level" attributes and a table with "root level" blocks. This was already described in #5390